### PR TITLE
fix: bugsnag error for emarsys track call

### PR DESCRIPTION
**Pull request by shrouti1507**
## What are the changes introduced in this PR?

Adding validation for eventsMapping config. If it is not at all configured, it is not coming as array

## What is the related Linear task?

Resolves INT-2204

## Please explain the objectives of your changes below

Put down any required details on the broader aspect of your changes. If there are any dependent changes, **mandatorily** mention them here

### Any changes to existing capabilities/behaviour, mention the reason & what are the changes ?

N/A

### Any new dependencies introduced with this change?

N/A

### Any new generic utility introduced or modified. Please explain the changes.

N/A

### Any technical or performance related pointers to consider with the change?

N/A

@coderabbitai review

<hr>

### Developer checklist

- [x] My code follows the style guidelines of this project

- [x] **No breaking changes are being introduced.**

- [x] All related docs linked with the PR?

- [x] All changes manually tested?

- [ ] Any documentation changes needed with this change?

- [x] Is the PR limited to 10 file changes?

- [x] Is the PR limited to one linear task?

- [x] Are relevant unit and component test-cases added?

### Reviewer checklist

- [ ] Is the type of change in the PR title appropriate as per the changes?

- [ ] Verified that there are no credentials or confidential data exposed with the changes.

### CODE REVIEW

**Comment by devops-github-rudderstack** [comment]
Test report for this run is available at: https://test-integrations-dev.s3.amazonaws.com/integrations-test-reports/rudder-transformer/3428/test-report.html

**Review by ItsSudip — APPROVED** [review]

**Review by anantjain45823 — APPROVED** [review]

File: test/integrations/destinations/emarsys/processor/data.ts [hunk 1]
```diff
@@ -1377,4 +1377,121 @@ export const data = [
       },
     },
   },
+  {
```
Comment (krishna2020) [inline test/integrations/destinations/emarsys/processor/data.ts:1380]: There's no need to write component test case for covering scenario of `eventsMapping` being null, It can be done by simple unit test around `deduceEventId`

Comment (krishna2020) [reply test/integrations/destinations/emarsys/processor/data.ts:1380]: I see we are failing the event itself, my bad - good with the test case
